### PR TITLE
Ports Delta_time

### PR DESCRIPTION
**Pull request by jupyterkat**
## About The Pull Request

ports
tgstation/tgstation#52981
and fixes

## Why It's Good For The Game

you should probably read the tg pr instead

🆑Donkie, Mothblocks
code: Process procs now properly utilize deltatime when implementing rates, timers and probabilities, which paves way for easier subsystem tickrate adjustments.
/🆑

### CODE REVIEW

**Comment by utilbird** [comment]
Our contributor guidelines will need to be updated for standardization if this is merged

**Comment by jupyterkat** [comment]
will do that

**Comment by utilbird** [comment]
> will do that

Nah, was mentioning it for maintainers.

**Comment by jupyterkat** [comment]
okay then

**Comment by MCterra10** [comment]
bookmarking for pre-emptive downstream updates
📑

**Comment by PowerfulBacon** [comment]
Cool

**Comment by bloons3** [comment]
@jupyterkat 
The recent lava burns stuff inside of it PR will need to be updated for delta time as well.
It was ported without it

**Comment by jupyterkat** [comment]
Already updated for delta time

**Comment by bloons3** [comment]
Blind copypaste isn't gonna work well.

DRUNK_HEALING is good having it's own process, but you need to remove the calls in Life() about it.
https://github.com/tgstation/tgstation/pull/46507

Double check that we have the Processing system for Quirks/Traits: https://github.com/tgstation/tgstation/pull/35440/files

**Comment by jupyterkat** [comment]
Not completely blind, but things go through and I can’t fix it all by myself.

**Comment by bloons3** [comment]
> Not completely blind, but things go through and I can’t fix it all by myself.

Add a checklist item to it in the main PR post

- [ ] task lists look like this
- [ ] and are good for keeping things on track
- [x] this one is completed

**Comment by bloons3** [comment]
No, don't remove the quirk Process() code.

Remove the DRUNK_HEALING quirk stuff from the Life() process

**Comment by jupyterkat** [comment]
No need, got most of it working, just tiny things needed to be fixed. I’ll deal with porting life()’s delta time later.

**Comment by jupyterkat** [comment]
I think the quirk process change should go with that pr. Thanks for pointing out

**Comment by Ghilker** [comment]
point of advice, don't add delta time to atmos stuff, we are removing it because it caused issues with the subsystem and pipelines

**Comment by jupyterkat** [comment]
@Ghilker pr not merged yet, I’ll port that later

**Comment by utilbird** [comment]
<strike>Draw straws for review assignment</strike>

**Comment by ike709** [comment]
@TheFakeElon you win

**Comment by utilbird** [comment]
> @TheFakeElon you win

😫

File: code/game/machinery/spaceheater.dm [hunk 1]
```diff
@@ -97,9 +97,9 @@
 
 		var/heat_capacity = env.heat_capacity()
 		var/requiredPower = abs(env.return_temperature() - targetTemperature) * heat_capacity
-		requiredPower = min(requiredPower, heatingPower)
+		requiredPower = min(requiredPower, heatingPower * delta_time)
 
-		if(requiredPower < 1)
+		if(requiredPower< 1)
```
Comment (utilbird) [inline code/game/machinery/spaceheater.dm]:
```suggestion
		if(requiredPower < 1)
```

File: code/modules/mob/living/silicon/pai/pai.dm [hunk 2]
```diff
@@ -74,7 +74,7 @@
 
 	var/emitterhealth = 20
 	var/emittermaxhealth = 20
-	var/emitterregen = 0.25
+	var/emitterregen = 1.25
```
Comment (utilbird) [inline code/modules/mob/living/silicon/pai/pai.dm]: This is a bit more than double 🤔

**Comment by utilbird** [comment]
Didn't see anything too out of place in the diffs, can't confirm if this accounts for all bee additions yet. I'll check tomorrow (probably)

**Review by utilbird — CHANGES_REQUESTED** [review]
-----  Important note: Most clockwork stuff doesn't have deltatime accounted for because TG removed it, I would recommend checking clockwork related objects and datums. I listed some below but not all -----

`alert.dm line` 434
`action.dm line` 655
`forced_movement.dm` line 37
`position_point_vector.dm` line 217
`rot.dm` line 38/48 (maybe, not sure about this one)
`wet_floor.dm` line 111. Might need to touch on SSwet_floors if nothing in this proc
`status_effect.dm` line 47
`areas.dm` line 379.
`dynamic.dm` line 583. midround injections use probability on process()
`meteor.dm` line 16. probs on process()

`revolution.dm` line 116/212.
`cloning.dm` line 276. (maybe)
`dance_machine.dm` line 459. prob checks
`race_converter.dm` line 84. prob checks
`airlock_types.dm` line 203. prob checks
`powersink.dm` line 122.
`weldingtool.dm` line 376.

`clockwork_massive.dm` line 61. prob checks
`clockwork_turfs.dm` line 270. prob checks
`brass_clothing.dm` line 131. prob checks
`interdiction_lens.dm` line 59. prob checks
`ocular_warden.dm` line 38.
`prosperity_prism.dm` line 56.
`clockwork_camera.dm` line 71.
`cult_structures.dm` line 169.
`nuclearbomb.dm` all processes.

**Comment by utilbird** [comment]
I've been through about half of the total process() calls now. I'll check the rest out later

There might be some false flags in there so I apologize for that in advance

**Comment by jupyterkat** [comment]
cockcult stuff converted

**Comment by jupyterkat** [comment]
all the delta_time'able stuff is done now

**Review by utilbird — DISMISSED** [review]
Contributor guidelines will need to be updated @ike709

**Review by utilbird — APPROVED** [review]
yolo

**Comment by utilbird** [comment]
shit. @PowerfulBacon need a tick from you

**Review by PowerfulBacon — APPROVED** [review]

**Comment by ike709** [comment]
Was this ever actually testmerged?

**Comment by jupyterkat** [comment]
none of you guys bothered to do it

**Comment by ike709** [comment]
Seems okay in TM. YOLO.

**Pull request merged**